Argo/Functions/decode.swift

```diff
@@ -1,35 +1,215 @@
-// MARK: decode root objects and arrays
+/**
+  Decode a raw JSON value into an optional `Decodable` value
```

Comment:
If this should be uppercase, there are a bunch of other lowercase "optional"s that should also be changed.